internal/utils/config.go

```diff
@@ -95,6 +96,10 @@ type (
 		Pop3Port uint `toml:"pop3_port"`
 	}
 
+	storage struct {
+		FileSizeLimit uint `toml:"file_size_limit_bytes"`
```

Comment:
```suggestion
		FileSizeLimit string `toml:"file_size_limit"`
```


For better human readability, may I suggest typing this field as string and use docker's [go-units](https://pkg.go.dev/github.com/docker/go-units) package to parse acceptable values like `50MiB` or `100KB`? Small usability improvements like this can add up to make a difference in terms of developer experience.

Reply:
I like the idea, and took a stab at implementing it. I broke my work into 2 commits:
- 8278165 - This is a minimal change that follows your suggestion of using a string field. However, this approach means we're not validating the string until it is used later (when starting containers), which feels like a bad experience.
- 012e632 - So, I took a stab at validating `file_size_limit` when parsing the toml ([here's an example I referenced in the toml package docs](https://github.com/BurntSushi/toml#using-the-marshaler-and-encodingtextunmarshaler-interfaces)). This means there's an extra struct and `Value` field -- but, I'm a novice with golang so perhaps you know how to make this cleaner.

Thoughts/suggestions?

Reply:
A few more things I wanted to point out:

1. The go-units package doesn't correctly handle binary sizes (https://github.com/docker/go-units/issues/31), but it's also not very actively maintained so this likely won't be fixed.
3. Supabase studio handles binary vs decimal byte sizes incorrectly (5MB is treated as 5MiB): https://github.com/supabase/supabase/blob/867fc0d739c145e8b48388ef2456157930e171d4/studio/tests/components/Storage/StorageSettings.utils.test.js#L13
4. The combination of the above 2 items means that if someone puts "5MB" into go-units (i.e. into the CLI config.toml) and "5MB" into supabase studio they'll get different values (5,000,000 vs 5,242,880).

Reply:
> So, I took a stab at validating file_size_limit when parsing the toml

This is a great idea. We can go a step further by using a type alias instead of a struct.

Reply:
> As a result, if someone puts "5MB" into go-units (i.e. into the CLI config.toml) and "5MB" into supabase studio they'll get different values (5,000,000 vs 5,242,880), which seems unideal.

Thanks for pointing out this difference. For compatibility with studio, we can use [RAMInBytes](https://pkg.go.dev/github.com/docker/go-units#RAMInBytes) for now.